### PR TITLE
MM-13740 Add additional plugin APIs for NPS plugin

### PR DESCRIPTION
**Pull request by hmhealey**
Exposes these fields so that we can include the following information with NPS results:
- Server version
- Server install date
- Server diagnostics ID
- License ID
- License SKU

Let me know if there's anywhere else that I should be documenting these plugin APIs or if this is sufficient.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-13740

### CODE REVIEW

File: app/plugin_api.go [hunk 1]
```diff
@@ -98,10 +98,22 @@ func (api *PluginAPI) SavePluginConfig(pluginConfig map[string]interface{}) *mod
...
 
+func (api *PluginAPI) GetLicense() *model.License {
+	return api.app.License()
+}
+
 func (api *PluginAPI) GetServerVersion() string {
 	return model.CurrentVersion
 }
 
+func (api *PluginAPI) GetSystemInstallDate() (int64, *model.AppError) {
```
Comment (hanzei) [inline app/plugin_api.go]: Out of interest: What is the use case for this?

File: app/plugin_api.go [hunk 2]
```diff
@@ -98,10 +98,22 @@ func (api *PluginAPI) SavePluginConfig(pluginConfig map[string]interface{}) *mod
 	return api.app.SaveConfig(cfg, true)
 }
 
+func (api *PluginAPI) GetLicense() *model.License {
```
Comment (hanzei) [inline app/plugin_api.go]: Out of interest: What is the use case for this?

File: app/plugin_api.go (hunk 1 above)
Comment (hmhealey) [reply app/plugin_api.go]: For both of these, they're to give some context to the Net Promoter Score survey results, so we can see if TE vs EE or how old the server is have an effect on how much users like Mattermost.

I don't know if the spec for the feature is public, but we're gathering [these fields](https://github.com/hmhealey/mattermost-plugin-nps/blob/master/server/segment.go#L54) with user feedback

Comment (hanzei) [reply app/plugin_api.go]: Makes sense, thanks for the explanation!

**Comment by esethna** [comment]
Docs to be generated programatically

**Pull request merged**